### PR TITLE
🎉 New Source: Aha [low-code cdk]

### PR DESCRIPTION
**Pull request by bala-ceg**
## What
Aha Integration: https://www.aha.io/api/

## How
Added the end points of features,releases,goals,users and ideas

## Recommended reading order
1. `spec.yaml`
2. `aha.yaml`
3.  `schemas\`

## 🚨 User Impact 🚨
No Impact. New Feature

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>
Screenshot 2022-11-03 at 12 17 40 PM

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hello! I'm going to be out of the office this Friday and won't be able to review your contribution again today, I return next Monday. So far, most contributions look solid and are almost done to be approved. As said in [Chris' comment](https://airbytehq.slack.com/archives/C0450PCSDB2/p1667430678866629) all contributions made before 2-November are eligible to receive the prize and have 2 weeks to merge the contributions. But I ensure next week we're going to have your contribution merged. If you have questions about the implementation you can send them in `#hacktoberfest-2022` in Airbyte's Slack.

Sorry the inconvenience and see you again next week, thank you so much for your contribution!

**Comment by bala-ceg** [comment]
Thanks  @marcosmarxm  for taking time to review this. DM'ed you on slack with test credentials and to the best of my knowledge I think I have all checklist items fulfilled 
	

- [x] Make sure you have added connector documentation to /docs/integrations/
- [x] Remove the file catalog from /integration_tests
- [x] Edit the sample_config.json inside /integration_tests
- [x] For the configured_catalog you can use only json_schema: {}
- [x] Add title to all properties in the spec.yaml
- [x ] Make sure the documentationUrl in the spec.yaml redirect to Airbyte's future connector page, eg: connector Airtable the documentationUrl: https://docs.airbyte.com/integrations/sources/airtable
- [x] Review now new line at EOF (end-of-file) for all files.

File: airbyte-integrations/connectors/source-aha/source_aha/schemas/TODO.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# TODO: Define your stream schemas
+Your connector must describe the schema of each stream it can output using [JSONSchema](https://json-schema.org). 
+
+You can describe the schema of your streams using one `.json` file per stream.
+
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-aha/source_aha/schemas/TODO.md:5]: Please remove this file.

File: airbyte-integrations/connectors/source-aha/source_aha/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  type: object
+  required:
+    - api_key
+    - url
+  additionalProperties: true
+  properties:
+    api_key:
+      type: string
+      description: API Key
+      title: API BEARER TOKEN
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-aha/source_aha/spec.yaml:14]: Please capitalize the title correctly: API Bearer Token

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-aha/source_aha/spec.yaml:14]: Same for the other parameter.

File: airbyte-integrations/connectors/source-aha/source_aha/aha.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      path: "/api/v1/features"
+  products_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "products"
+      path: "/api/v1/products"
+
+streams:
+  - "*ref(definitions.features_stream)"
+  - "*ref(definitions.products_stream)"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-aha/source_aha/aha.yaml:49]: Are you planning to add other endpoints?

File: airbyte-integrations/connectors/source-aha/source_aha/aha.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      $ref: "*ref(definitions.requester)"
+
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  features_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "features"
+      path: "/api/v1/features"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-aha/source_aha/aha.yaml:40]: Please move the prefix `/api/v1` to base URL.

File: airbyte-integrations/connectors/source-aha/source_aha/aha.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      type: PageIncrement
+      page_size: 5
+    page_token_option:
+      inject_into: "request_parameter"
+      field_name: "page"
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-aha/source_aha/aha.yaml:29]: you're not using the paginator, please change this.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments please check them.

File: airbyte-integrations/connectors/source-aha/source_aha/schemas/TODO.md (hunk 1 above)
Comment (bala-ceg) [reply airbyte-integrations/connectors/source-aha/source_aha/schemas/TODO.md]: Removed the file

File: airbyte-integrations/connectors/source-aha/source_aha/spec.yaml (hunk 2 above)
Comment (bala-ceg) [reply airbyte-integrations/connectors/source-aha/source_aha/spec.yaml]: Updated the title as suggested

File: airbyte-integrations/connectors/source-aha/source_aha/aha.yaml (hunk 3 above)
Comment (bala-ceg) [reply airbyte-integrations/connectors/source-aha/source_aha/aha.yaml:57]: added 3 more end points and verified the acceptance tests are passing

File: airbyte-integrations/connectors/source-aha/source_aha/aha.yaml (hunk 4 above)
Comment (bala-ceg) [reply airbyte-integrations/connectors/source-aha/source_aha/aha.yaml]: Moved it to base URL

File: airbyte-integrations/connectors/source-aha/source_aha/aha.yaml (hunk 5 above)
Comment (bala-ceg) [reply airbyte-integrations/connectors/source-aha/source_aha/aha.yaml]: Removed the paginator

**Comment by bala-ceg** [comment]
@marcosmarxm I addressed the review comments. Can you please review them?

File: airbyte-integrations/connectors/source-aha/source_aha/aha.yaml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  selector:
+    extractor:
+      field_pointer: []
+  requester:
+    url_base: "{{ config['url'] }}/api/v1"
+    http_method: "GET"
+    authenticator:
+      type: "BearerAuthenticator"
+      api_token: "{{ config['api_key'] }}"
+  increment_paginator:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-aha/source_aha/aha.yaml:13]: This is not used in the retriever. You must add this component to your base stream or retriever. Check other connectors how they do this.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
The paginator is not used yet.

File: airbyte-integrations/connectors/source-aha/source_aha/aha.yaml (hunk 6 above)
Comment (bala-ceg) [reply airbyte-integrations/connectors/source-aha/source_aha/aha.yaml:13]: Fixed this. Thanks @marcosmarxm  for pointing it out

Comment (bala-ceg) [reply airbyte-integrations/connectors/source-aha/source_aha/aha.yaml:13]: I raised the issue  [19274](https://github.com/airbytehq/airbyte/issues/19274) while testing the connector
Please let me know if my observations are valid

**Comment by bala-ceg** [comment]
> /test connector=connectors/source-aha
> 
> > 🕑 connectors/source-aha https://github.com/airbytehq/airbyte/actions/runs/3465751776
> > ❌ connectors/source-aha https://github.com/airbytehq/airbyte/actions/runs/3465751776
> > 🐛 https://gradle.com/s/kriorkzlxwzbm
> 
> ### Build Failed
> Test summary info:
> 
> ```
> =========================== short test summary info ============================
> FAILED test_core.py::TestBasicRead::test_read[inputs0] - AssertionError:  Rec...
> SKIPPED [1] ../usr/local/lib/python3.9/site-packages/source_acceptance_test/plugin.py:63: Skipping TestIncremental.test_two_sequential_reads: This connector does not implement incremental sync
> SKIPPED [1] ../usr/local/lib/python3.9/site-packages/source_acceptance_test/tests/test_core.py:51: The previous connector image could not be retrieved.
> SKIPPED [1] ../usr/local/lib/python3.9/site-packages/source_acceptance_test/tests/test_core.py:229: The previous connector image could not be retrieved.
> =================== 1 failed, 23 passed, 3 skipped in 45.88s ===================
> ```

Screenshot 2022-11-15 at 6 19 38 AM


@marcosmarxm in my local setup all tests are passing, can you please help me in fixing the test failures

**Review by marcosmarxm — APPROVED** [review]
thanks @bala-ceg

**Comment by bala-ceg** [comment]
Thanks @marcosmarxm  for helping me merge this connector

**Pull request merged**